Suggesting couple of changes to the document (#22298)
grafana/grafana: docs/sources/auth/gitlab.md

# GitLab OAuth2 Authentication

> To enable the GitLab OAuth2 you must register your application in GitLab. GitLab will generate a client ID and secret key for you to use.

Reviewer: I think the original wording of "an application in GitLab" is fine (it technically doesn't have to be _your_ application), but "GitLab OAuth2" sounds better than "the GitLab OAuth2".
```suggestion
To enable GitLab OAuth2 you must register an application in GitLab. GitLab will generate a client ID and secret key for you to use.
```

Author: But in all the previous OAuth2 documents, we have used your application while referring to application. In order to keep the similarity across all documents, we should change.

Reviewer: @oddlittlebird You think the consistency argument is strong enough to make the change? Personally I think the current wording makes sense from a technical POV.

Reviewer 2: We generally write instructions in the second person imperative, so aimed at "you do this." In this case, "you" is implied. If there are cases when a user (you) would be registering someone else's application, then you should use "an application" or "the application." Otherwise, "your application" is marginally better, because it keeps consistency with "you must..."

FWIW, I think "the application" makes more sense than "an application," because we are talking about a specific application that you want to enable OAuth2 for.

Reviewer: @oddlittlebird I think the technical side of things has to take precedence here over language considerations. The point of saying "an application" versus "your application" (or "the application") is that when you register an OAuth2 application with GitLab, it doesn't have to be any particular application, it just has to be configured correctly (i.e. the way we instruct the user). It's imperative that these technical instructions are followed correctly, or the integration won't work. How the instructions read from a language perspective has to be second priority, if you see what I mean?

Author: Have made changes to the patch as per recommendations, please merge

if you use the public `gitlab.com`, it means anyone in the world would be able
to login on your Grafana instance.

> You can however limit access to only members of a given group or list of

Reviewer 2: ```suggestion
You can limit access to only members of a given group or list of
```

Reviewer 2: Unnecessary words can be removed.
